twisted/internet/iocpreactor/tcp.py

```diff
@@ -59,7 +60,13 @@ def getHandle(self):
 
     def dataReceived(self, rbuffer):
         # XXX: some day, we'll have protocols that can handle raw buffers
-        self.protocol.dataReceived(str(rbuffer))
+        if not isinstance(rbuffer, bytes):
+            if _PY3:
```

Comment:
This dataReceived() call is nasty.  I went through an experiment of changing this
to not convert the argument at all, and just passthru the rbuffer argument to the next layer.  However, during the full test run, some of the tests failed, because they hit assertions elsewhere in the code where data was detected as type(unicode) (even on Python 2).

Some of those other protocols that pass unicode to write() methods need to be cleaned up,
but that is way out of the scope of this PR.